server/src/main/java/org/elasticsearch/index/mapper/BaseGeoShapeFieldMapper.java

```diff
@@ -0,0 +1,342 @@
+
+        private Builder getBuilder(String name, boolean coerce, boolean ignoreMalformed, Orientation orientation,
+                                   boolean ignoreZ, DeprecatedParameters deprecatedParameters) {
+            if (deprecatedParameters != null) {
+                return getLegacyBuilder(name, coerce, ignoreMalformed, orientation, ignoreZ, deprecatedParameters);
+            }
+            return new GeoShapeFieldMapper.Builder(name, coerce, ignoreMalformed, orientation, ignoreZ);
+        }
+
+        private Builder getLegacyBuilder(String name, boolean coerce, boolean ignoreMalformed, Orientation orientation,
+                                         boolean ignoreZ, DeprecatedParameters deprecatedParameters) {
+            deprecatedParameters.setup();
+            return new LegacyGeoShapeFieldMapper.Builder(name, coerce, ignoreMalformed, orientation, ignoreZ, deprecatedParameters);
+        }
+    }
+
+    public abstract static class BaseGeoShapeFieldType extends MappedFieldType {
+        protected Orientation orientation = Defaults.ORIENTATION.value();
+
+        protected BaseGeoShapeFieldType() {
+            setIndexOptions(IndexOptions.DOCS);
+            setTokenized(false);
+            setStored(false);
+            setStoreTermVectors(false);
+            setOmitNorms(true);
+        }
+
+        protected BaseGeoShapeFieldType(BaseGeoShapeFieldType ref) {
+            super(ref);
+            this.orientation = ref.orientation;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (!super.equals(o)) return false;
+            BaseGeoShapeFieldType that = (BaseGeoShapeFieldType) o;
+            return orientation == that.orientation;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), orientation);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public void checkCompatibility(MappedFieldType fieldType, List<String> conflicts) {
+            super.checkCompatibility(fieldType, conflicts);
+        }
+
+        public Orientation orientation() { return this.orientation; }
+
+        public void setOrientation(Orientation orientation) {
+            checkIfFrozen();
+            this.orientation = orientation;
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            return new TermQuery(new Term(FieldNamesFieldMapper.NAME, name()));
+        }
+
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            throw new QueryShardException(context, "Geo fields do not support exact searching, use dedicated geo queries instead");
+        }
+    }
+
+    protected Explicit<Boolean> coerce;
+    protected Explicit<Boolean> ignoreMalformed;
+    protected Explicit<Boolean> ignoreZValue;
+    protected Explicit<Orientation> orientation;
```

Comment:
Why do we hold the orientation here while it was only in the field type previously?

Reply:
Because orientation is used to build the geometry regardless of whether its S4J or Lucene. So it made more sense to have the parameter be part of the builder.